lm_eval/base.py

```diff
@@ -451,34 +456,121 @@ def higher_is_better(self):
 
     def fewshot_description(self):
         return ""
-
-    def fewshot_context(self, doc, num_fewshot, provide_description, rnd):
+    
+    def fewshot_context(self, doc, num_fewshot, provide_description, retrieval_method, retrieval_path, rnd):
         raw_description = self.fewshot_description()
         description = (raw_description + "\n===\n\n") if provide_description and raw_description else ""
 
         if num_fewshot == 0:
             labeled_examples = ""
         else:
-            # for sets with no training docs, draw from other set *but ensure no overlap with current doc*
-            if self.has_training_docs():
-                fewshotex = self.fewshot_examples(k=num_fewshot, rnd=rnd)
-            else:
-                if self._fewshot_docs is None:
-                    self._fewshot_docs = list(
-                        self.validation_docs() if self.has_validation_docs() else self.test_docs()
-                    )
+            if self.has_training_docs():         
+                if retrieval_method != 'random':
+                    # doc_idx = doc['doc_idx']
+                    # memory = self.read_memory(retrieval_path)
+                    # retrieval_idx = memory[doc_idx]['retrieval_idx']
+                    
+                    # ToDo: make an exception for which max_sim = 0.0 -> all nearest instances have 0.0 similarity
+                    try:
+                        selected_idx = self.sample_retreival_idx(doc, num_fewshot, retrieval_method, retrieval_path)
+                        fewshotex = self.fewshot_examples(k=num_fewshot, rnd=rnd, selected_idx=selected_idx)
+                    except:
+                        fewshotex = self.fewshot_examples(k=num_fewshot, rnd=rnd)
```

Comment:
```suggestion
                    except Exception as e:
                        fewshotex = self.fewshot_examples(k=num_fewshot, rnd=rnd)
                        logging.warning('exception in sample retreival: %s(%s)', type(e), str(e))
```

별건아닌데 원치 않는 로직이 돌았을때 혹시나 warning이라도 하나 찍어놓으면 나중에 sample_retreival_idx내부에서 실수로라도 코드가 일부 예외가 발생하더라도 문제상황을 알기 어려워서 요런거 추가해놓음 나중에 도움이 됩니다 ㅋ